tests/test_config.py

```diff
@@ -160,6 +160,27 @@ def test_uri_prioritised_over_host_and_port(self):
         assert mongo.db.name == 'database_name'
 
 
+    def test_missing_auth_mechanism_in_nonprefixed_config(self):
+
+        self.app.config["MONGO_HOST"] = 'localhost'
+        self.app.config["MONGO_PORT"] = 27017
+        self.app.config["MONGO_USERNAME"] = 'flask'
+        self.app.config["MONGO_PASSWORD"] = 'pymongo'
+        self.app.config['MONGO_DBNAME'] = 'test_db'
+
+        mongo = flask_pymongo.PyMongo(self.app)
+
+        assert mongo.db.name == 'test_db', 'wrong dbname: %s' % mongo.db.name
+
+        if pymongo.version_tuple[0] > 2:
```

Comment:
Is the `else` ever going to be hit? I suspect you mean `if pymongo.version_tuple[0] == 2:` here and L196?

EDIT: oh, right, I need coffee. this is a little confusing so I'd slightly prefer if you would make this ">= 3" to avoid my brain's bugs.